pkg/codegen/dotnet/gen.go

```diff
@@ -827,24 +872,35 @@ func (mod *modContext) genFunction(w io.Writer, fun *schema.Function) error {
 	return nil
 }
 
-func visitObjectTypes(t schema.Type, visitor func(*schema.ObjectType)) {
+func visitObjectTypesAcc(t schema.Type, visitor func(*schema.ObjectType), visited map[schema.Type]bool) {
```

Comment:
Not a big deal, but we've used `map[schema.Type]struct{}` for this in other languages.

See https://github.com/pulumi/pulumi/blob/f06e48b4140f5b72abdcc54b44bd736edc767952/pkg/codegen/go/gen.go#L874 for example

Reply:
You can also use `codegen.Set` if you don't need to iterate over the keys of the map.